src/query.rs

```diff
@@ -0,0 +1,60 @@
+use async_graphql::{Context, Error, Object, Result};
+use bson::Uuid;
+use s3::Bucket;
+
+use crate::{authentication::authenticate_user, media::Media, media_connection::MediaConnection};
+
+/// Describes GraphQL invoice queries.
+pub struct Query;
+
+#[Object]
+impl Query {
+    /// Entity resolver for media of specific id.
+    ///
+    /// Sets the path generated by MinIO behind PROXY_PATH.
+    /// Creates pre-signed paths with PATH_EXPIRATION_TIME.
+    async fn media_entity_resolver<'a>(
+        &self,
+        #[graphql(desc = "UUID of media file which a pre-signed path should be created for.")]
+        id: Uuid,
+    ) -> Result<Media> {
+        let media = Media { id: id };
+        Ok(media)
+    }
+
+    /// Returns pre-signed paths for a medias of UUIDs.
+    ///
+    /// Uses `self.get_media_path` to compute multiple media paths.
+    async fn get_media_ids<'a>(
```

Comment:
why is this called like this?
based on the return type I would have expected this to be called `medias`

Reply:
Yes, should be called `get_medias`, also need to fix the docs there